### ISSUE TITLE
Resampling error? - The behaviour of affine_transform with a one-dimensional array supplied for the matrix parameter has changed in scipy 0.18.0.

### ISSUE THREAD

**Issue by SunilKalmady** [nilearn/nilearn#1363]
I am trying to use NiftiLabelsMasker with an atlas and image (data) of different affine matrix.  I get very different time series when I set resampling_target = "labels" vs  resampling_target = "data".  Why does this happen?

from nilearn.input_data import NiftiLabelsMasker
masker = NiftiLabelsMasker(labels_img=atlas_filename, standardize="true",
                         verbose=5, resampling_target = "labels")
    
masker = NiftiLabelsMasker(labels_img=atlas_filename, standardize="true",
                           verbose=5, resampling_target = "data")

time_series = masker.fit_transform(fmri_filenames)

I believe I get correct results only when I set resampling_target = "labels", but this takes longer time.
Is it due to following warning ? -

> C:\Program Files\Anaconda3\lib\site-packages\scipy\ndimage\interpolation.py:430: UserWarning: The behaviour of affine_transform with a one-dimensional array supplied for the matrix parameter has changed in scipy 0.18.0.

I am using the most recent version of nilearn downloaded from github. Thanks.

**Comment by KamalakerDadi** [comment]
I used Nilearn provided atlas ```fetch_atlas_aal``` and functional datasets ```fetch_adhd```, seems like I get same timeseries.

There was scipy related problems with older Nilearn versions for instance in 0.2.5. This is issue is fixed with recent ones 0.2.6

Can you show us your timeseries ?

**Comment by SunilKalmady** [comment]
```
#Check version
import nilearn
nilearn.__version__
```

'0.2.6'

```
#Load atlas
from nilearn import datasets
dataset = datasets.fetch_atlas_aal(version='SPM12')
atlas_filename, labels = dataset.maps, dataset.labels

image.load_img(atlas_filename).affine

```
array([[  -2.,    0.,    0.,   90.],
       [   0.,    2.,    0., -126.],
       [   0.,    0.,    2.,  -72.],
       [   0.,    0.,    0.,    1.]])

```
#Load 4D nii data
from nilearn import image
#...
image.load_img(fmri_filenames).affine
```

array([[  -3.,    0.,    0.,   90.],
       [   0.,    3.,    0., -126.],
       [   0.,    0.,    3.,  -72.],
       [   0.,    0.,    0.,    1.]])

```
#Extract signals on a parcellation defined by labels - Resample images
from nilearn.input_data import NiftiLabelsMasker
masker = NiftiLabelsMasker(labels_img=atlas_filename, standardize="true",
                         verbose=5, resampling_target = "labels")
time_series = masker.fit_transform(fmri_filenames)
```

[NiftiLabelsMasker.fit_transform] loading data from C:\Users\kalmady/nilearn_data\aal_SPM12\aal\atlas\AAL.nii
[NiftiLabelsMasker.transform_single_imgs] Loading data from H:/gdrive/.../Filtered_4DVolume.nii
[NiftiLabelsMasker.transform_single_imgs] Resampling images
C:\Program Files\Anaconda3\lib\site-packages\scipy\ndimage\interpolation.py:430: UserWarning: The behaviour of affine_transform with a one-dimensional array supplied for the matrix parameter has changed in scipy 0.18.0.
  "The behaviour of affine_transform with a one-dimensional "
[NiftiLabelsMasker.transform_single_imgs] Extracting region signals
[NiftiLabelsMasker.transform_single_imgs] Cleaning extracted signals

```
#Plot ROI-50
plt.figure()
plt.plot(time_series [:,49])
plt.show()
```

![image](https://cloud.githubusercontent.com/assets/10730610/22272658/9e4a67fa-e258-11e6-849f-567855f6e704.png)

```
#Extract signals on a parcellation defined by labels - Resample atlas
masker = NiftiLabelsMasker(labels_img=atlas_filename, standardize="true",
                           verbose=5, resampling_target = "data")
time_series = masker.fit_transform(fmri_filenames)
```
[NiftiLabelsMasker.fit_transform] loading data from C:\Users\kalmady/nilearn_data\aal_SPM12\aal\atlas\AAL.nii
Resampling labels
[NiftiLabelsMasker.transform_single_imgs] Loading data from H:/gdrive/.../Filtered_4DVolume.nii
[NiftiLabelsMasker.transform_single_imgs] Extracting region signals
C:\Program Files\Anaconda3\lib\site-packages\scipy\ndimage\interpolation.py:430: UserWarning: The behaviour of affine_transform with a one-dimensional array supplied for the matrix parameter has changed in scipy 0.18.0.
  "The behaviour of affine_transform with a one-dimensional "
[NiftiLabelsMasker.transform_single_imgs] Cleaning extracted signals


```
#Plot ROI-50
plt.figure()
plt.plot(time_series [:,49])
plt.show()
```

![image](https://cloud.githubusercontent.com/assets/10730610/22272780/7fa2e056-e259-11e6-880e-879d348546f9.png)


I have looked into correlation matrices  across all ROIs to confirm that they are indeed very different between these two cases. 
Thanks for your reply.

**Comment by SunilKalmady** [comment]
Any answers here?

**Comment by GaelVaroquaux** [comment]
> Any answers here?

It's just a warning. If you use the latest nilearn, it's harmless as we
have taking this change of behavior in account. We cannot get rid of it,
as it comes from scipy, and not from nilearn.

**Comment by cancan101** [comment]
You can suppress this error like so: https://github.com/scipy/scipy/commit/d4aaed4a64682c331a6f87386e28f5b1784be2be.
